Clarify modelstructure for clocked variables (#2185)
modelica/fmi-standard: docs/2_2_common_mechanisms.adoc

Declaring a variable as <<clocked-variable,clocked variable>> using the <<clocks>> attribute declares that such a clocked variable can only be <<get-and-set-variable-values,accessed>> when one of the referenced Clocks is active.
This also holds for <<clocked-variable,clocked variables>> of type <<Clock>>.
The association between a clocked variable and its Clocks is expressed solely by the <<clocks>> attribute and must not be repeated as a <<dependencies,dependency>> in the <<ModelStructure>>.

[NOTE]
====
The <<clocks>> attribute and <<dependencies>> serve different purposes: <<clocks>> declares when a variable can be accessed, while <<dependencies>> declares algebraic dependencies on known variables.
The Clocks referenced by the <<clocks>> attribute and the variables listed in <<dependencies>> are therefore disjoint.
====

[NOTE]
====
Prior to FMI 3.0.3, the standard did not unambiguously define whether the association between a <<clocked-variable,clocked variable>> and its Clocks had to be repeated as a <<dependencies,dependency>> in the <<ModelStructure>>.

> This has been clarified in this version of the standard to allow only one possibility: this association is expressed solely by the <<clocks>> attribute and must not be listed as a <<dependencies,dependency>>.

Reviewer: Instead of "in this version", we maybe should specify the exact version "3.0.3" where this will be changed and this will stay for future versions of the document

When solving algebraic loops in <<EventMode>>, limitations to variable manipulations declared with attribute <<canHandleMultipleSetPerTimeInstant>> must be observed.

> A dependency declared in <<dependencies>> between <<clocked-variable,clocked variables>> only contributes to an algebraic loop while the variables involved can be accessed at the same super-dense time instant, that is, while their respective <<clocks>> are simultaneously active.

Reviewer: ```suggestion
A dependency declared in <<dependencies>> between <<clocked-variable,clocked variables>> contributes to an algebraic loop only when the variables involved can be accessed at the same super-dense time instant, that is, when their respective <<clocks>> are active at that super-dense time instant.
```

> Therefore, the importer must take the activation of the <<clocks>> declared via the <<clocks>> attribute into account when forming algebraic loop candidates across connected FMUs: a declared dependency between two clocked variables forms a loop candidate only if their Clocks can be active at the same super-dense time instant.

Reviewer: Do we need this normative sentence?
Isn't this actually an implementation suggestion?
I'd drop it, otherwise we need to define what is meant with loop candidate.
